nixos/modules/services/networking/onedrive.nix

```diff
@@ -0,0 +1,88 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.onedrive;
+
+  onedriveLauncher =  pkgs.writeShellScriptBin
+    "onedrive-launcher"
+    ''
+      # XDG_CONFIG_HOME is not recognized in the environment here.
+      if [ -f $HOME/.config/onedrive-launcher ]
+      then
+        # Hopefully using underscore boundary helps locate variables
+        for _onedrive_config_dirname_ in $(cat $HOME/.config/onedrive-launcher | grep -v '[ \t]*#' )
+        do
+          systemctl --user start onedrive@$_onedrive_config_dirname_
+        done
+      else
+        systemctl --user start onedrive@onedrive
+      fi
+    ''
+  ;
+
+  # Poor man's type/typo safety
+  systemdTargetList = [
+    "network-online"  
+    "multi-user"
+  ];
+
+  systemdTargets = builtins.listToAttrs
```

Comment:
this bit is needlessly complicated but it actually doesn't matter for reasons outlined below as it should just go.

Reply:
Left this in because your objection probably wasn't to this code in itself.